### PR TITLE
Refactor Hooks

### PR DESCRIPTION
**Pull request by wei2912**
This PR aims to refactor a couple of hooks within the codebase so as to make the code more idiomatic.

### CODE REVIEW

**Comment by wei2912** [comment]
The hash code is a bit off - the branch under `if (activeMentorId === null)` never runs as the initial state is `""`. Still experimenting with changes for that.

File: src/App.js [hunk 1]
```diff
@@ -76,14 +67,16 @@ function App() {
...
+  const [searchQuery, setSearchQuery] = useState({});
+  useEffect(() => setSearchValue(""), [waveIndex]);
+
+  const visibleMentorIds = useMemo(
+    () =>
+      searchValue.trim().length === 0
+        ? mentorIds[waveIndex]
+        : fieldSearch(searchQuery, waveIndex),
+    [searchQuery, searchValue, waveIndex]
+  );
```
Comment (bnjmnt4n) [inline src/App.js:79]: Wow this solution is pretty elegant. Guess I’m not too familiar with React’s hooks yet. 👍

**Comment by bnjmnt4n** [comment]
> The hash code is a bit off - the branch under `if (activeMentorId === null)` never runs as the initial state is `""`. Still experimenting with changes for that.
> 
> EDIT: Meh I guess we'll do that in another PR

Managed to get my computer working so I’ll look into this after this PR is merged :)

File: src/App.js (hunk 1 above)
Comment (wei2912) [reply src/App.js:79]: @bnjmnt4n Yeahh I learnt about the hooks during code review this week, which is why I thought of submitting this PR :joy:

**Pull request merged**